daqWheel/src/main.cpp

```diff
@@ -0,0 +1,117 @@
+#include <Arduino.h>
+#include "../lib/CAN/include/virtualTimer.h"
+#include "../lib/CAN/include/app_can.h"
+#include "pinDefs.h"
+
+#define SERIAL_DEBUG
+#define ARDUINO_TEENSY40
+
+#if defined(ARDUINO_TEENSY40) || defined(ARDUINO_TEENSY41)
+#include "teensy_can.h"
+// The bus number is a template argument for Teensy: TeensyCAN<bus_num>
+TeensyCAN<1> can_bus{};
+#endif
+
+#ifdef ARDUINO_ARCH_ESP32
+#include "esp_can.h"
+// The tx and rx pins are constructor arguments to ESPCan, which default to TX = 5, RX = 4
+ESPCan can_bus{};
+#endif
+
+//Structure for handling timers
+virtualTimerGroup_S read_timer;
+
+// TX CAN Message
+CANSignal<uint16_t, 0, 16, CANTemplateConvertFloat(0.1), CANTemplateConvertFloat(0), true> wheel_speed_signal{}; 
+CANSignal<uint16_t, 16, 16, CANTemplateConvertFloat(0.1), CANTemplateConvertFloat(-40), true> brake_temp_signal{}; 
+CANSignal<uint16_t, 32, 16, CANTemplateConvertFloat(1), CANTemplateConvertFloat(0), true> sus_pot_signal{}; 
+CANTXMessage<3> tx_message{can_bus, FL_CAN_FRAME_ADDRESS, 4, std::chrono::milliseconds{100}, wheel_speed_signal, brake_temp_signal, sus_pot_signal};
+
+// Wheel speed sensor
+unsigned long current_pulse_time = 0;
+unsigned long previous_pulse_time = 0;
+unsigned long pulse_duration = 0;
+
+/**
+ * @brief Runs on interrupt to record pulse duration
+ * 
+ */
+void ReadWheelSpeedSensorDuration()
+{
+    current_pulse_time = micros();
+    pulse_duration = current_pulse_time - previous_pulse_time;
+    previous_pulse_time = current_pulse_time;
+}
+
+/**
+ * @brief Uses the pulse duration to calculate the wheel speed in MPH
+ * 
+ */
+void ReadWheelSpeedSensor()
+{
+    // Wheel constants
+    const float kWheelCircumference = 1.26; //wheel circumference in m
+    const float kMPStoMPH = 2.24; // m/s to MPH constant 
+    const int kMagnetCount = 20;
+    //We multiply by 1M because our pulseDuration is measured in uS
+    float pulse_frequency = 1000000.0 / (float)pulse_duration;
+    float wheel_frequency = pulse_frequency / (float)kMagnetCount; 
+    wheel_speed_signal = (uint16_t)(wheel_frequency * kWheelCircumference * kMPStoMPH);
+}
+
+// Brake temperature
+/**
+ * @brief Reads the brake temperature sensor ADC value and converts it to temperature units (degrees C)
+ * 
+ */
+void ReadBrakeTempSensor()
+{
+  uint16_t raw_ADC_value = 0;
+  raw_ADC_value = analogRead(BRAKE_TEMPERATURE_SENSOR_PIN);
+  brake_temp_signal = raw_ADC_value * BRAKE_TEMPERATURE_SCALAR + BRAKE_TEMPERATURE_OFFSET;
+}
+
+// Suspension position sensor
+/**
+ * @brief Reads the suspension position sensor ADC value and converts it to length (mm)
+ * 
+ */
+void ReadSuspensionPositionSensor()
```

Comment:
Nitpick, but I would personally prefer we move all these read sensor functions into a separate sensors.c file or something like that to keep main a little cleaner. If that's too much effort, maybe just declaring them as prototypes at the top then actually defining them below the loop would be ok too, but I think styling conventions wants all private functions to come before all public function definitions, so maybe this is already compliant with recommendations.